functions/src/validation.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  body,
+  query,
+  ValidationError,
+  validationResult,
+} from "express-validator";
+import { NextFunction, Request, Response } from "express";
+import { APIFields, isValidUrl, ShareLinksCollection } from "./utils";
+import {
+  getShareLinkErrorByCode,
+  sendAndThrowShareLinkErrorCode,
+  ShareLinkErrorCode,
+} from "./error-handling";
+import * as functions from "firebase-functions";
+
+export function registerUrlValidationRules() {
```

Comment:
Not a huge deal, but out of curiosity, could these just be constants instead of functions, like:

```
export const registerUrlValidationRules = [
  body(ShareLinksCollection.URL).custom(isValidUrl),
  ...
];
```

If they do need to be functions, consider renaming to `getRegisterUrlValidationRules()` or similar since `registerUrlValidationRules()` sounds like it's registering validation rules. Same for `createApiKeyValidationRules()` etc.

Reply:
Ah, good point, they can be constants afaict!